Sources/NIOSOCKS/Channel Handlers/SOCKSClientHandler.swift

```diff
@@ -31,11 +31,16 @@ public final class SOCKSClientHandler: ChannelDuplexHandler {
     private var inboundBuffer: ByteBuffer?
     
     private var bufferedWrites: MarkedCircularBuffer<(NIOAny, EventLoopPromise<Void>?)> = .init(initialCapacity: 8)
+    private var establishHandler: (Channel) -> EventLoopFuture<Void>
     
     /// Creates a new `SOCKSClientHandler` that connects to a server
     /// and instructs the server to connect to `targetAddress`.
     /// - parameter targetAddress: The desired end point - note that only IPv4, IPv6, and FQDNs are supported.
-    public init(targetAddress: SOCKSAddress) {
+    public convenience init(targetAddress: SOCKSAddress) {
+        self.init(targetAddress: targetAddress, establishHandler: { $0.eventLoop.makeSucceededFuture(()) })
```

Comment:
```suggestion
        self.init(targetAddress: targetAddress, establishHandler: { $0.eventLoop.makeSucceededVoidFuture(()) })
```

Reply:
Is there any reasoning behind this, besides personal preference? I would assume that the compiler always uses `makeSucceededVoidFuture` given: https://github.com/apple/swift-nio/blob/main/Sources/NIO/EventLoop.swift#L602-L605 Personally I like to construct all my futures in exactly the same way, and assume that the compiler does the right thing.

And don't get me wrong happy to change. :) Just want to understand the reasoning...

Reply:
It's personal preference. The advantage of calling the specialized method is that you always call it: you don't rely on the optimiser seeing what you're doing and making the call go away. That's it. :)